windows/devices/uwb/include/windows/devices/uwb/IUwbDeviceDdi.hxx

```diff
@@ -32,46 +37,50 @@ struct IUwbDeviceDdi
-    virtual std::future<::uwb::protocol::fira::UwbStatus>
-    SessionRangingStart(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_STOP_RANGING_SESSION
-    virtual std::future<::uwb::protocol::fira::UwbStatus>
-    SessionRangingStop(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_GET_RANGING_COUNT
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::optional<uint32_t>>>
-    SessionGetRangingCount(uint32_t sessionId) = 0;
-
-    // IOCTL_UWB_SESSION_UPDATE_CONTROLLER_MULTICAST_LIST
-    virtual std::future<::uwb::protocol::fira::UwbSessionUpdateMulicastListStatus>
-    SessionUpdateControllerMulticastList(uint32_t sessionId, ::uwb::protocol::fira::UwbMulticastAction multicastAction, std::vector<::uwb::UwbMacAddress> controlees) = 0;
-
-    // IOCTL_UWB_GET_APP_CONFIG_PARAMS
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameter>>>
-    GetApplicationConfigurationParameters(uint32_t sessionId, std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameterType> applicationConfigurationParameterTypes) = 0;
-
-    // IOCTL_UWB_SET_APP_CONFIG_PARAMS
-    virtual std::future<std::tuple<::uwb::protocol::fira::UwbStatus, std::vector<::uwb::protocol::fira::UwbSetApplicationConfigurationParameterStatus>>>
-    SetApplicationConfigurationParameters(uint32_t sessionId, std::vector<::uwb::protocol::fira::UwbApplicationConfigurationParameter> applicationConfigurationParameters) = 0;
-
     // TODO: unspecified IOCTLs below
     // IOCTL_UWB_SET_DEVICE_CONFIG_PARAMS
     // IOCTL_UWB_GET_DEVICE_CONFIG_PARAMS
 };
+
+struct IUwbDeviceDdiConnector : public IUwbDeviceDdi
+{
+    /**
+     * @brief Get the name of this device.
+     *
+     * @return const std::string&
+     */
+    virtual const std::string&
+    DeviceName() const noexcept = 0;
+
+    /**
+     * @brief Sets the callbacks for the UwbDevice that owns this UwbDeviceConnector
+     *
+     * @param callbacks
+     * @return RegisteredCallbackToken* You can pass this pointer into DeregisterEventCallback to deregister this event callback
+     */
+    virtual ::uwb::RegisteredCallbackToken* RegisterDeviceEventCallbacks(std::weak_ptr<::uwb::UwbRegisteredDeviceEventCallbacks>) = 0;
+
+    /**
+     * @brief Start listening for notifications.
+     *
+     * Note: this is a rudimentary implementation and is only present to
+     * preserve existing behavior. It will eventually be replaced by a
+     * fine-grained publication/subscription model.
+     *
+     * @param onNotification The handler to invoke for each notification.
+     * @return true If listening for notifications started successfully.
+     * @return false If listening for notifications could not be started.
+     */
+    virtual bool
+    NotificationListenerStart() = 0;
+
+    /**
+     * @brief Stop listening for notifications.
+     */
+    virtual void
+    NotificationListenerStop() = 0;

```

Comment:
These should eventually be removed. The implementation should figure out when to start and stop listening based on the presence of callbacks. If it's easier, feel free to do that in a separate PR.